plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -739,6 +741,18 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession sess
         return Optional.empty();
     }
 
+    @Override
+    public void beginQuery(ConnectorSession session)
+    {
+        queryEventListeners.forEach(queryEventListener -> queryEventListener.beginQuery(session));
+    }
+
+    @Override
+    public void cleanupQuery(ConnectorSession session)
+    {
+        queryEventListeners.forEach(queryEventListener -> queryEventListener.cleanupQuery(session));
```

Comment:
A failure of one cleanupQuery should not prevent other listeners from receiving the cleanupQuery call.